pandas/io/parsers/python_parser.py

```diff
@@ -1166,6 +1164,28 @@ def _remove_skipped_rows(self, new_rows: list[list[Scalar]]) -> list[list[Scalar
             ]
         return new_rows
 
+    def _set_no_thousand_columns(self) -> set[int]:
+        no_thousands_columns: set[int] = set()
+        if self.columns and self.parse_dates:
+            assert self._col_indices is not None
+            no_thousands_columns = self._set_noconvert_dtype_columns(
+                self._col_indices, self.columns
+            )
+        if self.columns and self.dtype:
+            assert self._col_indices is not None
+            for i in self._col_indices:
+                if not isinstance(self.dtype, dict) and not is_numeric_dtype(
+                    self.dtype
+                ):
+                    no_thousands_columns.add(i)
+                if isinstance(self.dtype, dict) and not is_numeric_dtype(
+                    self.dtype.get(self.columns[i], None)
+                ):
+                    for key in self.dtype:
+                        if key == self.columns[i]:
+                            no_thousands_columns.add(i)
```

Comment:
Looks like the tests still pass without the
```python
                    for key in self.dtype:
                        if key == self.columns[i]:
```
, is there any reason to add it? I might be missing something, sorry

Reply:
All tests indeed pass without the
```
key in self.dtype:
    if key == self.columns[i]:
```
After adding the original example from [#50270](https://github.com/pandas-dev/pandas/issues/50270) tests failed. I noticed that it depends on whether we explicitly specify dtype for every column.
Like
`dtype={"a": str}`
Instead of
`dtype={"a": str, "b": np.int64, "c": np.int64}`

In this case, the check helps to pass tests. That's why I left the check.

Reply:
Not sure I follow, sorry - is there a way to write the test in such a way as that it would fail without the
```
for key in self.dtype:
    if key == self.columns[i]:
```
check? If so, could we add that as a test?